src/lib.rs

```diff
@@ -571,8 +571,12 @@ fn run_transport<F>(mut handler: F, transport: &mut Transport) where F: FnMut(Re
             Ok(sock) => sock,
             Err(e) => panic!(e.to_string()),
         };
+		let sp = match sock.peer() {
+			Ok(e)  => e,
+			Err(_) => return
```

Comment:
Returning from here would exit the loop anyway (and terminate the program). I'd rather let it panic with relevant error message.